### PR TITLE
Change default method to POST in emitter (close #289)

### LINKED ISSUE #289: Change default method to POST in emitter
As we have been doing on other trackers, we should also move to `POST` as the default method for making requests to the Collector.

I think there is one thing to consider before we do that: If using GET, the behaviour is intuitive – an event is sent right after it is tracked. If we move to POST with default buffer size 10, users might get confused because the tracker won't send any events if they track less than 10 events unless they call `flush()`. Also, since we only have an in-memory event queue, some events may get lost of users are not aware of the buffering. I can see several options on how to make this experience less confusing for users:

1. Set the default buffer size to 1
2. Document better that one needs to call flush when they want to see events sent to the collector and also before closing their app
3. Add a timer to the emitter that automatically flushes the events after some time (similar like mobile trackers but I don't think that this pattern is suitable for server-side apps)

I think that we should go with option 1 as that is the most intuitive when users are starting with the tracker. We should also document better how to make use of the buffer and how to call the flush. Opening this for discussion if there are other opinions.

### PR DESCRIPTION
**Pull request by Jack-Keene**
This PR sets the default method to post in the emitter as well as setting the Default buffer size to 1 (see discussion in [issue 289](https://github.com/snowplow/snowplow-python-tracker/issues/289)). It also updates the tests to reflect the expected changes.

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

File: snowplow_tracker/emitters.py [hunk 1]
```diff
@@ -332,7 +330,7 @@ def __init__(
             endpoint: str,
             protocol: HttpProtocol = "http",
             port: Optional[int] = None,
-            method: Method = "get",
+            method: Method = "post",
             buffer_size: Optional[int] = None,
```
Comment (matus-tomlein) [inline snowplow_tracker/emitters.py:334]: In the comment below, it says that the default buffer size is 10.

File: snowplow_tracker/emitters.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def __init__(
             endpoint: str,
             protocol: HttpProtocol = "https",
             port: Optional[int] = None,
-            method: Method = "get",
+            method: Method = "post",
```
Comment (matus-tomlein) [inline snowplow_tracker/emitters.py:56]: Can't comment in there directly but please also make sure to change the default method to POST in the `CeleryEmitter`, see `celery/celery_emitter.py`.

**Review by matus-tomlein — COMMENTED** [review]
Seems like there is an extra commit with changing the HTTPS default because we updated the release branch. You'll need to remove that from this branch, which is quite an annoying process, let me know if you need help.

**Review by matus-tomlein — COMMENTED** [review]
One more thing that I just realized is that the example app [makes use of an Emitter with default configuration](https://github.com/snowplow/snowplow-python-tracker/blob/master/examples/app.py#L15) – I suppose that won't track any events since there are less than 10 events tracked... Can you add a flush at the end?

**Review by matus-tomlein — APPROVED** [review]
LGTM!

**Pull request merged**